xbmc/windowing/gbm/drm/DRMAtomic.cpp

```diff
@@ -137,30 +137,36 @@ void CDRMAtomic::DrmAtomicCommit(int fb_id, int flags, bool rendered, bool video
     AddProperty(m_gui_plane, "CRTC_ID", 0);
   }
 
-  auto ret = drmModeAtomicCommit(m_fd, m_req, flags | DRM_MODE_ATOMIC_TEST_ONLY, nullptr);
+  auto ret = drmModeAtomicCommit(m_fd, m_req->Get(), flags | DRM_MODE_ATOMIC_TEST_ONLY, nullptr);
   if (ret < 0)
   {
     CLog::Log(LOGERROR, "CDRMAtomic::{} - test commit failed: {}", __FUNCTION__, strerror(errno));
+
+    UpdateRequestQueue(false);
+
+    // update the old atomic request with the new fb id to avoid tearing
+    if (rendered)
+      AddProperty(m_gui_plane, "FB_ID", fb_id);
   }
-  else if (ret == 0)
+
+  ret = drmModeAtomicCommit(m_fd, m_req->Get(), flags, nullptr);
+  if (ret < 0)
   {
-    ret = drmModeAtomicCommit(m_fd, m_req, flags, nullptr);
-    if (ret < 0)
-    {
-      CLog::Log(LOGERROR, "CDRMAtomic::{} - atomic commit failed: {}", __FUNCTION__,
-                strerror(errno));
-    }
+    CLog::Log(LOGERROR, "CDRMAtomic::{} - atomic commit failed: {}", __FUNCTION__,
+              strerror(errno));
   }
 
+  if (CServiceBroker::GetLogging().CanLogComponent(LOGWINDOWING))
+    m_req->LogAtomicRequest();
+
   if (flags & DRM_MODE_ATOMIC_ALLOW_MODESET)
   {
     if (drmModeDestroyPropertyBlob(m_fd, blob_id) != 0)
       CLog::Log(LOGERROR, "CDRMAtomic::{} - failed to destroy property blob: {}", __FUNCTION__,
                 strerror(errno));
   }
 
-  drmModeAtomicFree(m_req);
-  m_req = drmModeAtomicAlloc();
+  UpdateRequestQueue(true);
```

Comment:
I struggled a bit to find this initialization.
Imho it's not quite obvious that updating the queue initializes `m_req`.
How about explicitly initializing it?

Reply:
Yep, updated thanks!